lib/rules/lint-no-dom-events.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const Rule = require('./base');
+const AstNodeInfo = require('../helpers/ast-node-info');
+
+const ERROR_MESSAGE = 'Use Ember actions instead of DOM events.';
+
+module.exports = class NoDomEvents extends Rule {
+  visitor() {
+    return {
+      ElementNode(node) {
+        const domEventAttribute = node.attributes.find(isDomEventAttributeWithAction);
+        if (domEventAttribute) {
+          this.log({
+            message: ERROR_MESSAGE,
+            line: domEventAttribute.loc && domEventAttribute.loc.start.line,
+            column: domEventAttribute.loc && domEventAttribute.loc.start.column,
+            source: this.sourceForNode(domEventAttribute),
+          });
+        }
+      },
+    };
+  }
+};
+
+function isDomEventAttributeWithAction(node) {
+  return (
+    AstNodeInfo.isAttrNode(node) &&
+    node.name.startsWith('on') &&
```

Comment:
I'm not sure this is good enough 🤔, how certain are we that there are no legitimate attribute names starting with `on`?

Reply:
My understanding from looking at the [full list](https://www.w3schools.com/tags/ref_attributes.asp) of HTML attributes is that all event attributes and only event attributes begin with `on`. 

I implemented the rule with this attribute prefix check for simplicity so that the rule doesn't have to know the full list of nearly 100 [event attributes](https://www.w3schools.com/jsref/dom_obj_event.asp).

Alternatively, if I wanted to check for full attribute name matches, I could store the full event attribute list inside the rule (and hope the list doesn't change often), or get the list from another package like [event-attributes](https://www.npmjs.com/package/event-attributes) (although that package seems to have very few users).

Assuming my initial assumption is correct, the alternatives seem like overkill.